app/src/main/java/com/github/houseorganizer/houseorganizer/billsharer/Billsharer.java

```diff
@@ -0,0 +1,252 @@
+package com.github.houseorganizer.houseorganizer.billsharer;
+
+import android.util.Log;
+
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Billsharer {
+
+    private List<Expense> expenses;
+    private List<Debt> debts;
+    private Map<String, Integer> balances;
+    private final DocumentReference currentHouse;
+    private DocumentReference onlineReference;
+    private List<String> residents;
+    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+    public Billsharer(DocumentReference currentHouse) {
+        this.expenses = new ArrayList<>();
+        debts = new ArrayList<>();
+        this.currentHouse = currentHouse;
+        startUpBillsharer();
+    }
+
+    public Billsharer(DocumentReference currentHouse, DocumentReference onlineReference, List<Expense> expenses) {
+        this.expenses = expenses;
+        debts = new ArrayList<>();
+        this.currentHouse = currentHouse;
+        this.onlineReference = onlineReference;
+        startUpBillsharer();
+    }
+
+    private void startUpBillsharer() {
+        initResidents();
+        initBalances();
```

Comment:
I don't get how you can have the four calls back to back when they all depend on initResident which is asynchronous due to the Firebase request
Is there something i don't see or should they be in the onCompleteListener of initResidents ?

Reply:
right !